api/src/config.rs

```diff
@@ -0,0 +1,1520 @@
+        if &self.backend_type != "oss" {
+            Err(einval!("backend type is not 'OSS"))
+        } else {
+            self.oss
+                .as_ref()
+                .ok_or_else(|| einval!("no configuration information for OSS"))
+        }
+    }
+
+    /// Get configuration information for Registry
+    pub fn get_registry_config(&self) -> Result<&RegistryConfig> {
+        if &self.backend_type != "registry" {
+            Err(einval!("backend type is not 'registry'"))
+        } else {
+            self.registry
+                .as_ref()
+                .ok_or_else(|| einval!("no configuration information for registry"))
+        }
+    }
+}
+
+/// Configuration information for localfs storage backend.
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub struct LocalFsConfig {
+    /// Blob file to access.
+    #[serde(default)]
+    pub blob_file: String,
+    /// Dir to hold blob files. Used when 'blob_file' is not specified.
+    #[serde(default)]
+    pub dir: String,
+    /// Alternative dirs to search for blobs.
+    #[serde(default)]
+    pub alt_dirs: Vec<String>,
+}
+
+/// OSS configuration information to access blobs.
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub struct OssConfig {
+    /// Oss http scheme, either 'http' or 'https'
+    #[serde(default = "default_http_scheme")]
+    pub scheme: String,
+    /// Oss endpoint
+    pub endpoint: String,
+    /// Oss bucket name
+    pub bucket_name: String,
+    /// Prefix object_prefix to OSS object key, for example the simulation of subdirectory:
+    /// - object_key: sha256:xxx
+    /// - object_prefix: nydus/
+    /// - object_key with object_prefix: nydus/sha256:xxx
+    #[serde(default)]
+    pub object_prefix: String,
+    /// Oss access key
+    #[serde(default)]
+    pub access_key_id: String,
+    /// Oss secret
+    #[serde(default)]
+    pub access_key_secret: String,
+    /// Skip SSL certificate validation for HTTPS scheme.
+    #[serde(default)]
+    pub skip_verify: bool,
+    /// Drop the read request once http request timeout, in seconds.
+    #[serde(default = "default_http_timeout")]
+    pub timeout: u32,
+    /// Drop the read request once http connection timeout, in seconds.
+    #[serde(default = "default_http_timeout")]
+    pub connect_timeout: u32,
+    /// Retry count when read request failed.
+    #[serde(default)]
+    pub retry_limit: u8,
+    /// Enable HTTP proxy for the read request.
+    #[serde(default)]
+    pub proxy: ProxyConfig,
+    /// Enable mirrors for the read request.
+    #[serde(default)]
+    pub mirrors: Vec<MirrorConfig>,
+}
+
+/// Container registry configuration information to access blobs.
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub struct RegistryConfig {
+    /// Registry http scheme, either 'http' or 'https'
+    #[serde(default = "default_http_scheme")]
+    pub scheme: String,
```

Comment:
I remembered there was once a discussion about removing the `scheme` from the config file, just prefixing the host address with shame to construct a URL in the same style as contained. @imeoer .